agent/hcp/telemetry/otel_exporter.go

```diff
@@ -27,32 +27,32 @@ type EndpointProvider interface {
 	GetEndpoint() *url.URL
 }
 
-// OTELExporter is a custom implementation of a OTEL Metrics SDK metrics.Exporter.
+// otelExporter is a custom implementation of a OTEL Metrics SDK metrics.Exporter.
 // The exporter is used by a OTEL Metrics SDK PeriodicReader to export aggregated metrics.
 // This allows us to use a custom client - HCP authenticated MetricsClient.
-type OTELExporter struct {
+type otelExporter struct {
```

Comment:
Related discussion up in: https://github.com/hashicorp/consul/pull/18584#discussion_r1305870215

At some point in the future it might be used outside the package. But it isn't right now, and I subscribe to the school of YAGNI. If we actually wind up using this outside the package later I think we should switch it back, then

Reply:
Agree with the YAGNI approach. Good to also think about the public interface of this but considering it's changed a couple times now probably best that it's internal for the time being.

Reply:
+1 for thinking about the public interface if/when we have non-HCP consumers of this